src/redis-cli.c

```diff
@@ -8877,6 +9406,89 @@ static sds askPassword(const char *msg) {
     return auth;
 }
 
+/* Prints out the hint completion string for a given input prefix string. */
+void testHint(const char *input) {
+    cliInitHelp();
+
+    sds hint = getHintForInput(input);
+    printf("%s\n", hint);
+    exit(0);
+}
+
+sds readHintSuiteLine(char buf[], size_t size, FILE *fp) {
+    while (fgets(buf, size, fp) != NULL) {
+        if (buf[0] != '#') {
+            sds input = sdsnew(buf);
+
+            /* Strip newline. */
+            input = sdstrim(input, "\n");
+            return input;
+        }
+    }
+    return sdsnew("EOF");
+}
+
+/* Runs a suite of hint completion tests contained in a file. */
+void testHintSuite(char *filename) {
```

Comment:
Why isn't testHint() enough? I.e. can't the tests in TCL call `redis-cli --test_hint` for each of the hints to be tested? Would that be too slow?

Reply:
I honestly didn't try it that way. It was just convenient for me to have all the tests in one file. Again, whatever works is fine.

Reply:
Yes, the test suite txt file is really nice. I was just thinking maybe the test orchestration should be in the test, but it's no big deal. Or perhaps wrapped in an `#ifdef REDIS_TEST` or so. We do that in some other tests within c files, e.g. dict.